p2p/src/authenticated/discovery/actors/peer/actor.rs

```diff
@@ -70,15 +70,19 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng + Metrics, C: Pub
         )
     }
 
-    /// Unpack `msg` and verify the underlying `channel` is registered.
-    fn validate_msg<V>(msg: Option<Data>, rate_limits: &HashMap<u32, V>) -> Result<Data, Error> {
+    /// Unpack outbound `msg` and assert the underlying `channel` is registered.
+    fn validate_outbound_msg<V>(
+        msg: Option<Data>,
+        rate_limits: &HashMap<u32, V>,
+    ) -> Result<Data, Error> {
         let data = match msg {
             Some(data) => data,
             None => return Err(Error::PeerDisconnected),
         };
-        if !rate_limits.contains_key(&data.channel) {
-            return Err(Error::InvalidChannel);
-        }
+        assert!(
```

Comment:
This indicates a serious error in logic somewhere in p2p that should lead us to exit.

Reply:
Sure that's fine

> we should exit to avoid getting banned/blacklisted by peers

It did already short-circuit and not send the message, but I guess assert is fine if you prefer crashing

Reply:
We should never actually be able to hit this with the public interface, so I think it is worth asserting.